### PR TITLE
detect-dce-iface/v1: add tests

### PR DESCRIPTION
**Pull request by TheKharleeci**
Conversion: 

Dcerpc-dce-iface-02 : DetectDceIfaceTestParse1-2
dcerpc_udp.rs & dcerpc.rs: Dcerpc-dce-iface-02
Add test for Dcerpc-dce-stub-data
Add READme for Dcerpc-dce-iface-02

### CODE REVIEW

File: tests/dcerpc/dcerpc-dce-iface-02/test.rules [hunk 1]
```diff
@@ -1 +1,7 @@
-alert tcp any any -> any any (msg:"DCE Iface test";flow:established,to_server;dce_iface:afa8bd80-7d8a-11c9-bef4-08002b102989;sid:1;)
+alert tcp any any -> any any (msg:"DCE Iface test";flow:established,to_server;dcerpc.iface:afa8bd80-7d8a-11c9-bef4-08002b102989;sid:1;)
+alert tcp any any -> any any (msg:"DCE Iface test";flow:established,to_server;dcerpc.iface:afa8bd80-7d8a-11c9-bef4-08002b102989,=1; sid:2;)
+alert tcp any any -> any any (msg:"DCE Iface test";flow:established,to_server;dcerpc.iface:afa8bd80-7d8a-11c9-bef4-08002b102989,=1,any_frag; sid:3;)
+alert tcp any any -> any any (msg:"DCE opnum";flow:established,to_server; dcerpc.opnum:4; sid:4;)
+
+
+
```
Comment (inashivb) [inline tests/dcerpc/dcerpc-dce-iface-02/test.rules:7]: nit: too many newlines

File: tests/dcerpc/dcerpc-dce-iface-03/test.rules [hunk 2]
```diff
@@ -1 +1,2 @@
-alert tcp any any -> any any (msg:"DCE Iface test";flow:established,to_server;dce_iface:afa8bd80-7d8a-11c9-bef4-08002b102989,=1;sid:1;)
+alert tcp any any -> any any (msg:"DCE Iface test";flow:established,to_server;dcerpc.iface:afa8bd80-7d8a-11c9-bef4-08002b102989,=1;sid:1;)
+alert tcp any any -> any any (msg:"DCE opnum";flow:established,to_server; dcerpc.opnum:4; sid:2;)
```
Comment (inashivb) [inline tests/dcerpc/dcerpc-dce-iface-03/test.rules:2]: Could you separate out `opnum` tests to `dce-opnum` test?

File: tests/dcerpc/dcerpc-dce-iface-04/test.rules [hunk 3]
```diff
@@ -1 +1 @@
-alert tcp any any -> any any (msg:"DCE Iface test";flow:established,to_server;dce_iface:afa8bd80-7d8a-11c9-bef4-08002b102989,<1;sid:1;)
+alert tcp any any -> any any (msg:"DCE Iface test";flow:established,to_server;dcerpc.iface:afa8bd80-7d8a-11c9-bef4-08002b102989,<1;sid:1;)
```
Comment (inashivb) [inline tests/dcerpc/dcerpc-dce-iface-04/test.rules:1]: Could you please do a separate commit for changing keywords to new notations?

File: tests/dcerpc/dcerpc-dce-stub-data/test.rules [hunk 4]
```diff
@@ -0,0 +1,5 @@
+alert tcp any any -> any any (msg:"DCE Iface test";flow:established,to_server;dcerpc.iface:afa8bd80-7d8a-11c9-bef4-08002b102989;sid:1;)
+alert tcp any any -> any any (msg:"DCE stub data";flow:established,to_server; dcerpc.stub_data; content:"|09 00 00 00 00 01 00 00|"; sid:2;)
+
+
+
```
Comment (inashivb) [inline tests/dcerpc/dcerpc-dce-stub-data/test.rules:5]: nit: too many newlines

File: tests/dcerpc/dcerpc-dce-stub-data/test.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    - HAVE_LIBJANSSON
+
+args:
+- -k none
+
+checks:
+- filter:
+    count: 2
+    match:
+      event_type: dcerpc
```
Comment (inashivb) [inline tests/dcerpc/dcerpc-dce-stub-data/test.yaml:12]: We'd need a min-version check for 6 here since dcerpc logging wasn't supported before that.

File: tests/dcerpc/dcerpc-dce-stub-data/test.rules [hunk 6]
```diff
@@ -0,0 +1,5 @@
+alert tcp any any -> any any (msg:"DCE Iface test";flow:established,to_server;dcerpc.iface:afa8bd80-7d8a-11c9-bef4-08002b102989;sid:1;)
```
Comment (inashivb) [inline tests/dcerpc/dcerpc-dce-stub-data/test.rules:1]: Let's keep the rules for what they're supposed to do. Since this is not an iface test, and we already do the same test in `dce-iface-02`, let's skip this.

**Review by inashivb — CHANGES_REQUESTED** [review]

**Comment by TheKharleeci** [comment]
Followed By: #752
